src/Xrefcheck/Verify.hs

```diff
@@ -241,20 +246,43 @@ instance Given ColorMode => Buildable VerifyError where
       #{err}
       |]
 
-    PermanentRedirectError url Nothing ->
+    RedirectChainCycle chain ->
       [int||
-      Permanent redirect found:
-      #{url}
+      Cycle found in the following redirect chain:
+      #{interpolateIndentF 2 $ attachToRedirectChain chain "here"}
       |]
 
-    PermanentRedirectError url (Just redirectedUrl) ->
+    RedirectChainBroken chain ->
```

Comment:
Maybe `RedirectMissingLocation`? `RedirectChainBroken` sounds too general, not saying what is broken.